pkg/provider/aws/resources/elasticache.go

```diff
@@ -0,0 +1,91 @@
+package resources
+
+import (
+	"fmt"
+
+	"github.com/klothoplatform/klotho/pkg/config"
+	"github.com/klothoplatform/klotho/pkg/core"
+)
+
+type (
+	Elasticache struct {
+		Name            string
+		Engine          string
+		CloudwatchGroup *LogGroup
+		SubnetGroup     *ElasticacheSubnetgroup
+		SecurityGroups  []*SecurityGroup
+		ConstructsRef   []core.AnnotationKey
+	}
+
+	ElasticacheSubnetgroup struct {
+		Name          string
+		Subnets       []*Subnet
+		ConstructsRef []core.AnnotationKey
+	}
+)
+
+const (
+	EC_TYPE   = "elasticache"
+	ECSN_TYPE = "elasticache_subnetgroup"
+)
+
+// Provider returns name of the provider the resource is correlated to
+func (ec *Elasticache) Provider() string {
+	return AWS_PROVIDER
+}
+
+// KlothoResource returns AnnotationKey of the klotho resource the cloud resource is correlated to
+func (ec *Elasticache) KlothoConstructRef() []core.AnnotationKey {
+	return ec.ConstructsRef
+}
+
+// ID returns the id of the cloud resource
+func (ec *Elasticache) Id() string {
+	return fmt.Sprintf("%s:%s:%s", ec.Provider(), EC_TYPE, ec.Name)
+}
+
+// Provider returns name of the provider the resource is correlated to
+func (ecsn *ElasticacheSubnetgroup) Provider() string {
+	return AWS_PROVIDER
+}
+
+// KlothoResource returns AnnotationKey of the klotho resource the cloud resource is correlated to
+func (ecsn *ElasticacheSubnetgroup) KlothoConstructRef() []core.AnnotationKey {
+	return ecsn.ConstructsRef
+}
+
+// ID returns the id of the cloud resource
+func (ecsn *ElasticacheSubnetgroup) Id() string {
+	return fmt.Sprintf("%s:%s:%s", ecsn.Provider(), ECSN_TYPE, ecsn.Name)
+}
+
+func CreateElasticache(cfg *config.Application, dag *core.ResourceGraph, source core.Construct) *Elasticache {
+	ec := &Elasticache{
+		Name:            source.Id(),
+		Engine:          "redis", // TODO determine this from the type of `source`
+		CloudwatchGroup: NewLogGroup(cfg.AppName, fmt.Sprintf("/aws/elasticache/%s-%s-persist-redis", cfg.AppName, source.Id()), source.Provenance(), 0),
+		SubnetGroup: &ElasticacheSubnetgroup{
+			Name:          source.Id(),
+			Subnets:       GetSubnets(cfg, dag),
+			ConstructsRef: []core.AnnotationKey{source.Provenance()},
+		},
+		SecurityGroups: []*SecurityGroup{GetSecurityGroup(cfg, dag)},
```

Comment:
I still dont love the GetSubnets and GetSecurityGroup because its not specific to the network and boxes us in to only being able to support a single vpc.

I wont consider this blocking for now but i do think we should rewrite this logic

Reply:
We don't know how multiple networks will work, so it's hard to say how this would look like (does it take in the VPC? just the specific subnets?). I added a TODO to mark that it will need to change at some point.